### DISCUSSION TITLE
Feature Request: Populate `targetlist` for pg `Plan*`

### DISCUSSION THREAD

**Discussion by dentiny** [duckdb/pg_duckdb#463] [Ideas]
Why do you want this feature?
I'm working on a `RETURNING` clause for [pg_mooncake](https://github.com/Mooncake-Labs/pg_mooncake), after some debugging I found is `Plan*` [generated by pg_duckdb](https://github.com/Mooncake-Labs/pg_mooncake/blob/12b39ea3af35218fab0f9c4d8e0ecdcb5be4dc8c/src/pgduckdb/pgduckdb_planner.cpp#L54-L105) doesn't include [`targetlist` data member](https://github.com/postgres/postgres/blob/5d39becf8ba0080c98fee4b63575552f6800b012/src/include/nodes/plannodes.h#L153), which leads to 0 `natts` thus no [printed](https://github.com/postgres/postgres/blob/5d39becf8ba0080c98fee4b63575552f6800b012/src/backend/access/common/printtup.c#L303-L304) value.

After manually setting the `targetlist` to be
```
postgres_plan->planTree = duckdb_plan;
postgres_plan->planTree->targetlist = postgres_plan->planTree->targetlist;
```
it works in **some** cases.

Another observation, `PlannedStmt*` created by postgres does contain `targetlist`, not gets [overwritten](https://github.com/duckdb/pg_duckdb/blob/c3e80d8c27cc681380cf87b0e1f1ea95da87dcad/src/pgduckdb_planner.cpp#L131-L147) by duckdb plan. I expect the only difference between pg plan vs pg_duckdb plan to be execution plan, but not target list.

**Comment by JelteF** [comment]
This should be addressed by #468

**Reply by dentiny** [reply]
Thank you! I will close the feature request and wait for the PR to checkin :)
